google/datalab/utils/_http.py

```diff
@@ -44,17 +44,15 @@ def __init__(self, status, content):
     self.message = 'HTTP request failed'
     # Try extract a message from the body; swallow possible resulting ValueErrors and KeyErrors.
     try:
-      if type(content) == str:
-        error = json.loads(content)['error']
-      else:
-        error = json.loads(str(content, encoding='UTF-8'))['error']
+      error = json.loads(content, encoding='UTF-8')['error']
       if 'errors' in error:
         error = error['errors'][0]
       self.message += ': ' + error['message']
     except Exception:
-      lines = content.split('\n') if isinstance(content, basestring) else []
+      lines = content.splitlines() if isinstance(content, basestring) else []
```

Comment:
Looks like this is the only needed fix here, why are the lines below needed?

Reply:
Can't concat (+) bytes to str if content is bytes in python3.